pillar.example

```diff
@@ -2,39 +2,27 @@
 
 nomad:
   bin_dir: /usr/bin
-  config_dir: /etc/nomad
+  config_dir: /etc/nomad.d
   config:
     data_dir: /var/lib/nomad
     # Nodes not bound to consul must be configured to advertise themselves.
-    advertise:
-      http: "{%- for interface, ips in grains['ip4_interfaces'].items()
+    bind_addr: "{%- for interface, ips in grains['ip4_interfaces'].items()
                   if interface not in ['lo', 'cni0', 'docker0'] and
                   not interface.startswith('veth') and
                   ips|length > 0 %}
                   {%- if loop.first %}
                     {{- ips[0] -}}
                   {%- endif %}
-             {%- endfor %}"
-      rpc: "{%- for interface, ips in grains['ip4_interfaces'].items()
-                  if interface not in ['lo', 'cni0', 'docker0'] and
-                  not interface.startswith('veth') and
-                  ips|length > 0 %}
-                  {%- if loop.first %}
-                    {{- ips[0] -}}
-                  {%- endif %}
-            {%- endfor %}"
-      serf: "{%- for interface, ips in grains['ip4_interfaces'].items()
-                  if interface not in ['lo', 'cni0', 'docker0'] and
-                  not interface.startswith('veth') and
-                  ips|length > 0 %}
-                  {%- if loop.first %}
-                    {{- ips[0] -}}
-                  {%- endif %}
-             {%- endfor %}"
+                {%- endfor %}"
     server:
       enabled: True
       bootstrap_expect: 1
       encrypt: "AaABbB+CcCdDdEeeFFfggG=="
+      server_join:
+        retry_join:
+        {% for _, addrs in salt.saltutil.runner('mine.get', tgt='*', fun='network.ip_addrs') | dictsort() %}
+          - {{ addrs[0] }}
+        {% endfor %}
```

Comment:
@rbjorklin I think this addition is preventing Nomad agent to start.

The function must be allowed to call from Salt "mine" explicitly, but the tests seems do not do that.
So, when there are nothing rendered within the for loop, the `retry_join` value would be `None`, which is invalid I suppose.
I suggest you to check if there any values come out from the `mine.get` call first, and only then iterate over them to render complete `server_join` dictionary.